flink-table/flink-table-storage/src/main/java/org/apache/flink/table/storage/filestore/lsm/sst/SstFile.java

```diff
@@ -0,0 +1,230 @@
+            this.reader = reader;
+        }
+
+        @Nullable
+        @Override
+        public RecordIterator<KeyValue> readBatch() throws IOException {
+            BulkFormat.RecordIterator<RowData> next = reader.readBatch();
+            return next == null ? null : new SstFileRecordIterator(next);
+        }
+
+        @Override
+        public void close() throws IOException {
+            reader.close();
+        }
+    }
+
+    private class SstFileRecordIterator implements RecordIterator<KeyValue> {
+        BulkFormat.RecordIterator<RowData> iterator;
+
+        private SstFileRecordIterator(BulkFormat.RecordIterator<RowData> iterator) {
+            this.iterator = iterator;
+        }
+
+        @Override
+        public boolean advanceNext() throws IOException {
+            RecordAndPosition<RowData> next = iterator.next();
+            if (next == null) {
+                return false;
+            }
+
+            KeyValueSerializer tmp = serializer;
+            serializer = previousSerializer;
+            previousSerializer = tmp;
+
+            serializer.fromRow(rowDataSerializer.copy(next.getRecord()));
+            return true;
+        }
+
+        @Override
+        public boolean singleInstance() {
+            return false;
+        }
+
+        @Override
+        public KeyValue current() {
+            return serializer.getReusedKv();
+        }
+
+        @Override
+        public void releaseBatch() {
+            iterator.releaseBatch();
+        }
+    }
+
+    private class RollingFile {
+        private final Path actualPath;
+        private final long suggestedFileSize;
+
+        private final FSDataOutputStream out;
+        private final BulkWriter<RowData> writer;
+        private final RowDataSerializer keySerializer;
+        private final FieldStatsCollector statsCollector;
+
+        private long rowCount;
+        private BinaryRowData minKey;
+        private RowData maxKey;
+        private long minSequenceNumber;
+        private long maxSequenceNumber;
+
+        private RollingFile(Path path, long suggestedFileSize, int idx) throws IOException {
+            this.actualPath = new Path(path.toString() + "." + idx);
+            this.suggestedFileSize = suggestedFileSize;
+
+            this.out =
+                    actualPath
+                            .getFileSystem()
+                            .create(actualPath, FileSystem.WriteMode.NO_OVERWRITE);
+            this.writer = writerFactory.create(out);
+            this.keySerializer = new RowDataSerializer(keyType);
+            this.statsCollector = new FieldStatsCollector(valueType);
+
+            this.rowCount = 0;
+            this.minKey = null;
+            this.maxKey = null;
+            this.minSequenceNumber = Long.MAX_VALUE;
+            this.maxSequenceNumber = Long.MIN_VALUE;
+        }
+
+        private boolean write(KeyValue kv) throws IOException {
```

Comment:
Instead of return boolean, how about this class provide a method `boolean reachTargetSize()`.
Return false is weird. Is the element inserted successfully, is false full or is true full?